karoo_gp/base_class.py

```diff
@@ -1787,1535 +2043,1588 @@ class labels provided in the data .csv. Outputs an array of tuples
 
-        If however a function mutates into a terminal, the entire branch
-        beneath the function is deleted from the array and the entire array is
-        updated, to fix parent/child links, associated arities, and node IDs.
+    Called by: fx_nextgen_branch_mutate
 
-        Called by: fx_nextgen_branch_mutate
+    Arguments required: tree, branch
+    '''
 
-        Arguments required: tree, branch
-        '''
+    # replaces 2 instances, below; tested 2016 07/09
+    branch_top = int(branch.root[0])
+    # 'tree_depth_max' - depth at 'branch_top' to set max potential
+    # size of new branch - 2016 07/10
+    branch_depth = tree_depth_max - int(tree.root[4][branch_top])
+    if branch_depth < 0:  # this has never occured ... yet
+        error(f'\n\t\033[31m ERROR! In fx_evolve_grow_mutate: branch_depth '
+              f'{branch_depth} < 0')
+        # # consider special instructions for this (pause) - 2019 06/08
+        # self.fx_karoo_pause()
+
+    # the point of mutation ('branch_top') chosen resides at the maximum
+    # allowable depth, so mutate term to term
+    elif branch_depth == 0:
+
+        log(f'\t\033[36m max depth branch node\033[1m '
+            f'{tree.root[3][branch_top]} \033[0;0m\033[36mmutates '
+            f'from \033[0;0m\033[36mmutates from \033[1mterm\033[0;0m '
+            '\033[36mto \033[1mterm\033[0;0m\n', display=['i'])
+        log(f'\n\n\033[33m *** Grow Mutation: terminal to terminal '
+            f'*** \033[0;0m\n\n\033[36m This is the unaltered '
+            f'tourn_winner:\033[0;0m\n {tree.root}', display=['db'])
 
-        # replaces 2 instances, below; tested 2016 07/09
-        branch_top = int(branch[0])
-        # 'tree_depth_max' - depth at 'branch_top' to set max potential
-        # size of new branch - 2016 07/10
-        branch_depth = self.tree_depth_max - int(tree[4][branch_top])
-
-        if branch_depth < 0:  # this has never occured ... yet
-            print('\n\t\033[31m ERROR! In fx_evolve_grow_mutate: branch_depth',
-                  branch_depth, '< 0')
-            # consider special instructions for this (pause) - 2019 06/08
-            self.fx_karoo_pause()
-
-        # the point of mutation ('branch_top') chosen resides at the maximum
-        # allowable depth, so mutate term to term
-        elif branch_depth == 0:
-
-            self.log('\t\033[36m max depth branch node\033[1m '
-                     f'{tree[3][branch_top]} \033[0;0m\033[36mmutates '
-                     'from \033[0;0m\033[36mmutates from \033[1mterm\033[0;0m '
-                     '\033[36mto \033[1mterm\033[0;0m\n', display=['i'])
-            self.log('\n\n\033[33m *** Grow Mutation: terminal to terminal '
-                     '*** \033[0;0m\n\n\033[36m This is the unaltered '
-                     f'tourn_winner:\033[0;0m\n {tree}', display=['db'])
+        # call the previously loaded .csv which contains all terminals
+        rnd = rng.integers(0, len(terminals) - 1)
```

Comment:
Same as above -- are we sure that the `-1` is correct here?
```suggestion
        rnd = rng.integers(len(terminals) - 1)
```